src/JointTrajectoryControllerBase.cpp

```diff
@@ -202,10 +211,22 @@ void JointTrajectoryControllerBase::updateStep(const ros::Time& time,
     trajectory->evaluateDerivative(timeFromStart, 2, mDesiredAcceleration);
 
     // TODO: Check path constraints.
-    // TODO: Check goal constraints.
+
+    // Check goal constraints.
+    bool goalConstraintsSatisfied = true;
+    for (const auto& pair : mGoalConstraints) {
+      auto dof = mSkeleton->getDof(pair.first);
+      if (dof) {
+        std::size_t index = dof->getIndexInSkeleton();
+        if (std::abs(mDesiredPosition[index] - mActualPosition[index]) > pair.second) {
+          goalConstraintsSatisfied = false;
+          break;
+        }
+      }
+    }
 
     // Terminate the current trajectory.
-    if (timeFromStart >= trajectory->getDuration()) {
+    if (timeFromStart >= trajectory->getDuration() && goalConstraintsSatisfied) {
```

Comment:
Should we add a comment that we're not looking at `constraints/goal_time`?

Alternatively, maybe we should just document which parameters we actually support.

Reply:
Alright, but where is the best place to document it? A comment at this line seems redundant.